Assets/_PlatformerDevelopment/Tests/StateManagerTests.cs

```diff
@@ -76,6 +89,27 @@ void OnStateChanged(State state)
             //Assert
             Assert.AreEqual(resultingState, State.Play, "Resulting state should be Play");
         }
+        
+        [Test]
+        public void TestStateManagerCallsBackOnStateChangeFromAnotherStateManagerObject()
+        {
+            //Arrange
+            var result = 0;
+            void OnStateChanged(State state)
+            {
+                result++;
+            }
+
+            var stateManager = StateManager.Instance;
+            var stateManagerOther = StateManager.Instance;
+            stateManager.OnStateChanged += OnStateChanged;
```

Comment:
These 3 lines is the bug. 
Before:
`var stateManager = new StateManager()`
`var stateManagerOther = new StateManager()`

each one had its own OnStateChange(). So subscribing to one, does not subscribe it to the other.